modules/core/src/main/scala/com/github/scoquelin/arugula/commands/internal/LettuceRedisCommandDelegation.scala

```diff
@@ -0,0 +1,29 @@
+package com.github.scoquelin.arugula.commands.internal
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+ * A trait that provides a delegation mechanism for Lettuce Redis commands
+ * to be dispatched and lifted into Scala Futures from the Lettuce RedisFuture
+ *
+ * @tparam K The type of the Redis key
+ * @tparam V The type of the Redis value
+ */
+private[commands] trait LettuceRedisCommandDelegation[K, V] {
+  /**
+   * Lifts the outcome (RedisFuture[T]) of a Lettuce Redis command into a Scala Future[T]
+   *
+   * @param command The Lettuce command to be dispatched
+   * @tparam T The type of the returned value inside the io.lettuce.core.RedisFuture[T]
+   * @return A Scala Future[T] extracted from the io.lettuce.core.RedisFuture[T]
+   */
+  private[commands] def delegateRedisClusterCommandAndLift[T](
+    command: io.lettuce.core.cluster.api.async.RedisClusterAsyncCommands[K, V] => io.lettuce.core.RedisFuture[T]
+  ): Future[T]
+
+  /**
+   * The execution context
+   * Should be injected by the implementing class
+   */
+  private[commands] implicit val executionContext: ExecutionContext
```

Comment:
I don't love how I have to define the `executionContext` this way and override it. Feels clunky. but not sure how else to do it without creating a big mess everywhere else.

Reply:
Yeah that's what I was looking into quickly... could not come up with anything more elegant. We would need [trait parameters](https://docs.scala-lang.org/scala3/reference/other-new-features/trait-parameters.html) maybe from Scala 3